graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -127,34 +129,23 @@
 from .validation import validate_schema_and_query_ast
 
 
-# LocationStackEntry contains the following:
-# - location: Location object corresponding to an inserted MarkLocation block
-# - num_traverses: Int counter for the number of traverses inserted after the last MarkLocation
-#                  (corresponding Location stored in `location`)
-LocationStackEntry = namedtuple("LocationStackEntry", ("location", "num_traverses"))
+class OutputMetadata(NamedTuple):
+    """Metadata about a query's outputs.
 
+    Containing the following members:
+        type: GraphQL type object, such as GraphQLString or GraphQLInt, describing the type of the
+              output value.
+        optional: boolean, whether the output is part of an optional traversal, which would allow a
+                  value of null.
+        folded: boolean, whether the output is within a fold scope. Note that the "type" is expected
+                to be a GraphQLList unless the output field is an _x_count, in which case the type
+                must be GraphQLInt.
+    """
 
-def _construct_location_stack_entry(location, num_traverses):
-    """Return a LocationStackEntry namedtuple with the specified parameters."""
-    if not isinstance(num_traverses, int) or num_traverses < 0:
-        raise AssertionError(
-            u"Attempted to create a LocationStackEntry namedtuple with an invalid "
-            u'value for "num_traverses" {}. This is not allowed.'.format(num_traverses)
-        )
-    if not isinstance(location, Location):
-        raise AssertionError(
-            u"Attempted to create a LocationStackEntry namedtuple with an invalid "
-            u'value for "location" {}. This is not allowed.'.format(location)
-        )
-    return LocationStackEntry(location=location, num_traverses=num_traverses)
```

Comment:
Was this dead code and removed intentionally? If so, neat!

Reply:
Seemed dead to me! I went to convert `LocationStackEntry` to a `NamedTuple` and realized nothing is calling this code 😄